### PR TITLE
Remove incidental_inventory_foreman test.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

Remove incidental_inventory_foreman test.

##### ISSUE TYPE

Test Pull Request

##### COMPONENT NAME

incidental_inventory_foreman integration test

### CODE REVIEW

File: .azure-pipelines/azure-pipelines.yml [hunk 1]
```diff
@@ -183,8 +183,6 @@ stages:
               test: ios/csr1000v/
             - name: VyOS Python
               test: vyos/1.1.8/
-            - name: Cloud Python
-              test: cloud/
```
Comment (s-hertel) [inline .azure-pipelines/azure-pipelines.yml:187]: Will ansible-test-cloud-aws still run? It uses the cloud/aws alias: https://github.com/ansible/ansible/blob/devel/test/integration/targets/ansible-test-cloud-aws/aliases

Comment (mattclay) [reply .azure-pipelines/azure-pipelines.yml:187]: Yes, it runs under the Generic tests, not the Incidental tests.

Here's last night's run: https://dev.azure.com/ansible/ansible/_build/results?buildId=44602&view=logs&j=ab74475c-3143-5b70-01a7-2ace1ea1c05f&t=c892da64-2ffc-519e-f0d1-23758c705c71&l=3036

Comment (s-hertel) [reply .azure-pipelines/azure-pipelines.yml:187]: Oh, I totally missed the `stage: Incidental` above

Comment (mattclay) [reply .azure-pipelines/azure-pipelines.yml:187]: No problem. Thanks for checking.

**Pull request merged**